[Dashboard] Inject / extract tag references

nickpeihl opened the pull request:
Fixes #210619

## Summary

Provides a tags array on the request and response bodies of dashboards.

This allows consumers of the Dashboards HTTP API and internal RPC API to add an array of tag names to the attributes in the body of create and update endpoints. The dashboard server will be responsible for converting the tag names into references in the saved object. 

If, during creation or update, a tag name does not have a matching tag saved object, a new tag saved object will be created. If the user lacks permissions to manage tags, then an error will be logged in the server and the tag will not be added to the dashboard.

The server also injects the tag references as an array of tag names in the attributes of the response body of get and search endpoints of the HTTP and RPC APIs. 

For backwards compatibility in create and update endpoints, tags can alternatively be specified in the `references` array in the options instead of (or in addition to) the `attributes.tags` in the request body. Similarly, for backwards compatibility, tag references are returned in the `references` of the response body of get and search endpoints.

Client-side tag handling is out of scope for this PR. Dashboards listing page and dashboard settings continue to use the tag references and do not use the `tags` attribute in the response.

For example:

Here's how we currently create a dashboard with tag references.
```
## Creating a dashboard with tag references
POST kbn:/api/dashboards/dashboard
{
  "attributes": {
    "title": "my tagged dashboard"
  },
  "references": [
    {
      "type": "tag",
      "id": "37aab5de-a34d-47cb-9aa5-9375d5db595f",
      "name": "tag-ref-37aab5de-a34d-47cb-9aa5-9375d5db595f"
    },
    {
      "type": "tag",
      "id": "5ed29bba-c14d-4302-9a8c-9602e40dbc2a",
      "name": "tag-ref-5ed29bba-c14d-4302-9a8c-9602e40dbc2a"
    },
    {
      "type": "tag",
      "id": "fc7890e8-c00f-44a1-88a2-250e4d27e61d",
      "name": "tag-ref-fc7890e8-c00f-44a1-88a2-250e4d27e61d"
    }
  ]
}
```

With this PR, creating a dashboard with tags is much simpler.

```
## Creating a dashboard with tag names
POST kbn:/api/dashboards/dashboard
{
  "attributes": {
    "title": "my tagged dashboard",
    "tags": [
      "boo",
      "foo",
      "bingo",
      "bongo"
    ]
  }
}

```

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] If there are more than one tag saved objects with the same name, only one of the tag references will be added to the saved object when creating a dashboard. Creating tags with duplicate names are not permitted via the UI. But there is no such restrictions when creating tags from imported saved objects. Having multiple tags with the same name is an edge case that Kibana guards against with reasonable restrictions, so I think we should not be too concerned about it.
- [ ] ...





<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

nickpeihl committed:
Inject / extract tag references

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

nickpeihl committed:
Fixes

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/notice'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/capture_oas_snapshot --include-path /api/status --include-path /api/alerting/rule/ --include-path /api/alerting/rules --include-path /api/actions --include-path /api/security/role --include-path /api/spaces --include-path /api/streams --include-path /api/fleet --include-path /api/dashboards --update'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285183)
* Commit: 16e760e2ade8ff71bac593a8293f03f9521a0ce8

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/285183#0195aaf3-4a64-474b-a4c9-5b89b9b0761a)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285183","number":285183,"commit":"16e760e2ade8ff71bac593a8293f03f9521a0ce8"}],"number":285183}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

nickpeihl committed:
Merge branch 'main' into refHandling-tags

nickpeihl committed:
Add api integration tests with tags

nickpeihl committed:
Create tag if saved object matching tag name not found

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/286283)
* Commit: 3095adb5afdb27594ec9d2219067f3ec495f7aa0

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/286283#0195ba3e-230b-43ac-adb9-f528475061f8)

### History
* :broken_heart: [Build #285368](https://buildkite.com/elastic/kibana-pull-request/builds/285368) failed 89b1312e862462b6c16428f2ee8a50a9fc9233e1
* :broken_heart: [Build #285183](https://buildkite.com/elastic/kibana-pull-request/builds/285183) failed 16e760e2ade8ff71bac593a8293f03f9521a0ce8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286283","number":286283,"commit":"3095adb5afdb27594ec9d2219067f3ec495f7aa0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285368","number":285368,"commit":"89b1312e862462b6c16428f2ee8a50a9fc9233e1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285183","number":285183,"commit":"16e760e2ade8ff71bac593a8293f03f9521a0ce8"}],"number":286283}
buildkite-pr-comment-->

nickpeihl committed:
Merge branch 'main' of https://github.com/elastic/kibana into refHandling-tags

TinaHeiligers reviewed (approved):
Core-related changes (test fixtures) LGTM.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/286710)
* Commit: bfe95bd54e48af83467866f0e147200fd6051ab1

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/286710#0195c901-a3b3-4484-961a-ff6aec7ea0fe)

### History
* :broken_heart: [Build #286283](https://buildkite.com/elastic/kibana-pull-request/builds/286283) failed 3095adb5afdb27594ec9d2219067f3ec495f7aa0
* :broken_heart: [Build #285368](https://buildkite.com/elastic/kibana-pull-request/builds/285368) failed 89b1312e862462b6c16428f2ee8a50a9fc9233e1
* :broken_heart: [Build #285183](https://buildkite.com/elastic/kibana-pull-request/builds/285183) failed 16e760e2ade8ff71bac593a8293f03f9521a0ce8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286710","number":286710,"commit":"bfe95bd54e48af83467866f0e147200fd6051ab1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286283","number":286283,"commit":"3095adb5afdb27594ec9d2219067f3ec495f7aa0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285368","number":285368,"commit":"89b1312e862462b6c16428f2ee8a50a9fc9233e1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285183","number":285183,"commit":"16e760e2ade8ff71bac593a8293f03f9521a0ce8"}],"number":286710}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/kibana-presentation (Team:Presentation)

ThomThomson commented on src/platform/plugins/shared/dashboard/public/dashboard_api/get_serialized_state.ts:
```diff
@@ -138,6 +138,8 @@ export const getSerializedState = ({
     { embeddablePersistableStateService: embeddableService }
   );
 
+  // TODO Provide tags as an array of tag names in the attribute. In that case, tag references
```
Do we have a followup issue for this?

nickpeihl replied:
https://github.com/elastic/kibana/issues/216346

ThomThomson commented on src/platform/plugins/shared/dashboard/server/content_management/v2/cm_services.ts:
```diff
@@ -37,7 +37,7 @@ export const serviceDefinition: ServicesDefinition = {
       data: {
         schema: dashboardAttributesSchema,
-        up: (data: DashboardCrudTypes['CreateIn']['data']) => attributesTov3(data),
+        up: (data: DashboardCrudTypes['CreateIn']['data']) => attributesTov3(data, [], () => []),
```
Why do we pass a blank references array here?

nickpeihl replied:
Ah, those are now optional parameters. The content management `up` transforms do not supply `references` so we only transform the attributes. Fixed in ff8e17a77ec737fbbcc160b1636904dc9f49660f.

ThomThomson commented on src/platform/test/api_integration/apis/dashboards/create_dashboard/main.ts:
```diff
@@ -171,6 +172,153 @@ export default function ({ getService }: FtrProviderContext) {
       expect(response.body.item.attributes.panels).to.be.an('array');
     });
 
+    describe('create a dashboard with tags', () => {
```
Great to see a new API Integration test added!

ThomThomson reviewed (approved):
Great work! Tested this locally in chrome by creating and updating a Dashboard SO via the API through devtools. Also tested that tags generated this way were treated the same + looked the same in the SO Management page as tags created via the UI.

Looked through the code and left a few questions, but otherwise everything looks pretty straightforward. Lots of test coverage added, which is very nice!

nickpeihl committed:
Fix unit test

nickpeihl committed:
Move tag creation to wrapper function

This allows saved object migrations to continue to use the synchronous itemAttrsToSavedObject function since they don't have a need for tag creation.

nickpeihl committed:
Remove optional params from up transform

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/288167)
* Commit: ff8e17a77ec737fbbcc160b1636904dc9f49660f

### Failed CI Steps
* [Linting (with types)](https://buildkite.com/elastic/kibana-pull-request/builds/288167#0195dd91-104e-42b6-8303-78fe6ad039ea)

### History
* :broken_heart: [Build #286710](https://buildkite.com/elastic/kibana-pull-request/builds/286710) failed bfe95bd54e48af83467866f0e147200fd6051ab1
* :broken_heart: [Build #286283](https://buildkite.com/elastic/kibana-pull-request/builds/286283) failed 3095adb5afdb27594ec9d2219067f3ec495f7aa0
* :broken_heart: [Build #285368](https://buildkite.com/elastic/kibana-pull-request/builds/285368) failed 89b1312e862462b6c16428f2ee8a50a9fc9233e1
* :broken_heart: [Build #285183](https://buildkite.com/elastic/kibana-pull-request/builds/285183) failed 16e760e2ade8ff71bac593a8293f03f9521a0ce8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288167","number":288167,"commit":"ff8e17a77ec737fbbcc160b1636904dc9f49660f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286710","number":286710,"commit":"bfe95bd54e48af83467866f0e147200fd6051ab1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286283","number":286283,"commit":"3095adb5afdb27594ec9d2219067f3ec495f7aa0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285368","number":285368,"commit":"89b1312e862462b6c16428f2ee8a50a9fc9233e1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285183","number":285183,"commit":"16e760e2ade8ff71bac593a8293f03f9521a0ce8"}],"number":288167}
buildkite-pr-comment-->

nickpeihl committed:
Merge branch 'main' into refHandling-tags

nickpeihl committed:
Fix lint error

dominiqueclarke commented:
Will this require updating to a new version of the content management api? We're currently using version 3.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/288201)
* Commit: b1645c7d803af12da379490f41a75035b08902f3

### Failed CI Steps
* [FTR Configs #17](https://buildkite.com/elastic/kibana-pull-request/builds/288201#0195ddec-7d44-4b27-92d5-417d493d192b)
* [FTR Configs #17](https://buildkite.com/elastic/kibana-pull-request/builds/288201#0195de06-90b5-4cb4-9e3e-ad0c786cf48f)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288201#0195ddec-7d44-4b27-92d5-417d493d192b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288201/jobs/0195ddec-7d44-4b27-92d5-417d493d192b/artifacts/0195de06-7819-4fee-9381-1ec0c4f9171e) FTR Configs #<span></span>17 / apis dashboards dashboards - update main update a dashboard with tags empty tags array removes all tags
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288201#0195de06-90b5-4cb4-9e3e-ad0c786cf48f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288201/jobs/0195de06-90b5-4cb4-9e3e-ad0c786cf48f/artifacts/0195de0c-cd5c-4b66-98ff-f27d2c719146) FTR Configs #<span></span>17 / apis dashboards dashboards - update main update a dashboard with tags empty tags array removes all tags

### History
* :broken_heart: [Build #288167](https://buildkite.com/elastic/kibana-pull-request/builds/288167) failed ff8e17a77ec737fbbcc160b1636904dc9f49660f
* :broken_heart: [Build #286710](https://buildkite.com/elastic/kibana-pull-request/builds/286710) failed bfe95bd54e48af83467866f0e147200fd6051ab1
* :broken_heart: [Build #286283](https://buildkite.com/elastic/kibana-pull-request/builds/286283) failed 3095adb5afdb27594ec9d2219067f3ec495f7aa0
* :broken_heart: [Build #285368](https://buildkite.com/elastic/kibana-pull-request/builds/285368) failed 89b1312e862462b6c16428f2ee8a50a9fc9233e1
* :broken_heart: [Build #285183](https://buildkite.com/elastic/kibana-pull-request/builds/285183) failed 16e760e2ade8ff71bac593a8293f03f9521a0ce8

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288201","number":288201,"commit":"b1645c7d803af12da379490f41a75035b08902f3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288167","number":288167,"commit":"ff8e17a77ec737fbbcc160b1636904dc9f49660f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286710","number":286710,"commit":"bfe95bd54e48af83467866f0e147200fd6051ab1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286283","number":286283,"commit":"3095adb5afdb27594ec9d2219067f3ec495f7aa0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285368","number":285368,"commit":"89b1312e862462b6c16428f2ee8a50a9fc9233e1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285183","number":285183,"commit":"16e760e2ade8ff71bac593a8293f03f9521a0ce8"}],"number":288201}
buildkite-pr-comment-->

nickpeihl committed:
Fix functional test failure

nickpeihl committed:
Update functional test fixture

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/288298)
* Commit: fc9e0e7e5cc807deeb683cf16cbbf1d91b4f2b4a

### Failed CI Steps
* [FTR Configs #106](https://buildkite.com/elastic/kibana-pull-request/builds/288298#0195e295-a32e-4b37-b08d-b60326621b7b)
* [FTR Configs #106](https://buildkite.com/elastic/kibana-pull-request/builds/288298#0195e2b1-45a6-4d01-9f45-eead4f82fefa)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288298#0195e295-a32e-4b37-b08d-b60326621b7b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288298/jobs/0195e295-a32e-4b37-b08d-b60326621b7b/artifacts/0195e2b1-34a5-48c4-9344-af7ada563da8) FTR Configs #<span></span>106 / apis dashboards dashboards - list main should retrieve a paginated list of dashboards
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288298#0195e2b1-45a6-4d01-9f45-eead4f82fefa) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288298/jobs/0195e2b1-45a6-4d01-9f45-eead4f82fefa/artifacts/0195e2b7-1bea-438f-8fb3-c2993e77b56e) FTR Configs #<span></span>106 / apis dashboards dashboards - list main should retrieve a paginated list of dashboards

### History
* :broken_heart: [Build #288225](https://buildkite.com/elastic/kibana-pull-request/builds/288225) failed aee41999e6fffd23147dd14ff2008edfe0f3b256
* :broken_heart: [Build #288201](https://buildkite.com/elastic/kibana-pull-request/builds/288201) failed b1645c7d803af12da379490f41a75035b08902f3
* :broken_heart: [Build #288167](https://buildkite.com/elastic/kibana-pull-request/builds/288167) failed ff8e17a77ec737fbbcc160b1636904dc9f49660f
* :broken_heart: [Build #286710](https://buildkite.com/elastic/kibana-pull-request/builds/286710) failed bfe95bd54e48af83467866f0e147200fd6051ab1
* :broken_heart: [Build #286283](https://buildkite.com/elastic/kibana-pull-request/builds/286283) failed 3095adb5afdb27594ec9d2219067f3ec495f7aa0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288298","number":288298,"commit":"fc9e0e7e5cc807deeb683cf16cbbf1d91b4f2b4a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288225","number":288225,"commit":"aee41999e6fffd23147dd14ff2008edfe0f3b256"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288201","number":288201,"commit":"b1645c7d803af12da379490f41a75035b08902f3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288167","number":288167,"commit":"ff8e17a77ec737fbbcc160b1636904dc9f49660f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286710","number":286710,"commit":"bfe95bd54e48af83467866f0e147200fd6051ab1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286283","number":286283,"commit":"3095adb5afdb27594ec9d2219067f3ec495f7aa0"}],"number":288298}
buildkite-pr-comment-->

nickpeihl committed:
Fix test

nickpeihl committed:
Merge branch 'main' into refHandling-tags

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/288548)
* Commit: c4362c322dd2c02aae32afde9e0e7fb07ab2b57e

### Failed CI Steps
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/288548#0195ec63-bf74-4ee0-810f-83de15839378)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/288548#0195ec63-bf74-4ee0-810f-83de15839378) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/288548/jobs/0195ec63-bf74-4ee0-810f-83de15839378/artifacts/0195ec8a-2537-409b-8d05-f2ec47a528a1) Jest Tests #<span></span>4 / TemplatesList renders template details correctly

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #288298](https://buildkite.com/elastic/kibana-pull-request/builds/288298) failed fc9e0e7e5cc807deeb683cf16cbbf1d91b4f2b4a
* :broken_heart: [Build #288225](https://buildkite.com/elastic/kibana-pull-request/builds/288225) failed aee41999e6fffd23147dd14ff2008edfe0f3b256
* :broken_heart: [Build #288201](https://buildkite.com/elastic/kibana-pull-request/builds/288201) failed b1645c7d803af12da379490f41a75035b08902f3
* :broken_heart: [Build #288167](https://buildkite.com/elastic/kibana-pull-request/builds/288167) failed ff8e17a77ec737fbbcc160b1636904dc9f49660f
* :broken_heart: [Build #286710](https://buildkite.com/elastic/kibana-pull-request/builds/286710) failed bfe95bd54e48af83467866f0e147200fd6051ab1
* :broken_heart: [Build #286283](https://buildkite.com/elastic/kibana-pull-request/builds/286283) failed 3095adb5afdb27594ec9d2219067f3ec495f7aa0

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288548","number":288548,"commit":"c4362c322dd2c02aae32afde9e0e7fb07ab2b57e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288298","number":288298,"commit":"fc9e0e7e5cc807deeb683cf16cbbf1d91b4f2b4a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288225","number":288225,"commit":"aee41999e6fffd23147dd14ff2008edfe0f3b256"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288201","number":288201,"commit":"b1645c7d803af12da379490f41a75035b08902f3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/288167","number":288167,"commit":"ff8e17a77ec737fbbcc160b1636904dc9f49660f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/286710","number":286710,"commit":"bfe95bd54e48af83467866f0e147200fd6051ab1"}],"number":288548}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/14333621427

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 8.x:<br> - [[Dashboard] Remove mSearch from content management (#210709)](https://github.com/elastic/kibana/pull/210709)<br> - [[Dashboards] Provide a method for fetching dashboards on the Dashboard server plugin contract (#209678)](https://github.com/elastic/kibana/pull/209678)|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 214788
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.x|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 8.x:<br> - [[Dashboard] Remove mSearch from content management (#210709)](https://github.com/elastic/kibana/pull/210709)<br> - [[Dashboards] Provide a method for fetching dashboards on the Dashboard server plugin contract (#209678)](https://github.com/elastic/kibana/pull/209678)|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 214788
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

nickpeihl commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/217590">](https://github.com/elastic/kibana/pull/217590)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
Looks like this PR has a backport PR but it still hasn't been merged. Please merge it ASAP to keep the branches relatively in sync.
